### PR TITLE
Issue 56: Fix sa_crypto_cipher_process_last processing

### PR DESCRIPTION
**Pull request by ericeberry**

### CODE REVIEW

**Review by riwoh — DISMISSED** [review]

File: reference/src/client/test/sa_key_unwrap_aes_gcm.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ namespace {
         ASSERT_NE(wrapping_key, nullptr);
 
         std::vector<uint8_t> aad = random(1024);
-        std::vector<uint8_t> tag = random(AES_BLOCK_SIZE);
```
Comment (mhabrat) [inline reference/src/client/test/sa_key_unwrap_aes_gcm.cpp:132]: Update copyright year to include 2023. Please check other files to ensure copyright year is updated.

File: reference/src/taimpl/src/ta_sa_crypto_cipher_process_last.c [hunk 2]
```diff
@@ -579,6 +628,13 @@ sa_status ta_sa_crypto_cipher_process_last(
                 ERROR("ta_sa_crypto_cipher_process_last_aes_ctr failed");
                 break;
             }
+        } else if (cipher_algorithm == SA_CIPHER_ALGORITHM_CHACHA20) {
+            status = ta_sa_crypto_cipher_process_last_chacha20(out_bytes, cipher, in_bytes, bytes_to_process,
+                    caller_uuid);
+            if (status != SA_STATUS_OK) {
+                ERROR("ta_sa_crypto_cipher_process_last_aes_ctr failed");
```
Comment (mhabrat) [inline reference/src/taimpl/src/ta_sa_crypto_cipher_process_last.c:635]: ta_sa_crypto_cipher_process_last_chacha20

**Review by mhabrat — CHANGES_REQUESTED** [review]

File: reference/src/client/test/sa_key_unwrap_aes_gcm.cpp (hunk 1 above)
Comment (ericeberry) [reply reference/src/client/test/sa_key_unwrap_aes_gcm.cpp:132]: Done

File: reference/src/taimpl/src/ta_sa_crypto_cipher_process_last.c (hunk 2 above)
Comment (ericeberry) [reply reference/src/taimpl/src/ta_sa_crypto_cipher_process_last.c:635]: Done

**Review by mhabrat — APPROVED** [review]

**Review by mhabrat — APPROVED** [review]

**Pull request merged**